components/common/src/templating/mod.rs

```diff
@@ -91,6 +94,50 @@ impl TemplateRenderer {
             .render(template, &raw)
             .map_err(|e| Error::TemplateRenderError(format!("{}", e)))
     }
+
+    // This method is only implemented so we can intercept the call to Handlebars and display
+    // a deprecation message to users. More information here https://github.com/habitat-sh/habitat/issues/6323.
+    // When Handlebars is upgraded this can be safely removed.
+    pub fn register_template_file<P>(&mut self,
+                                     name: &str,
+                                     path: P)
+                                     -> result::Result<(), TemplateFileError>
+        where P: AsRef<std::path::Path>
+    {
+        let path = path.as_ref();
+
+        lazy_static! {
+            static ref RE: Regex =
+                Regex::new(r"(\{\{[^}]+[^.])(\[[^}]*\}\})").expect("Failed to compile template \
+                                                                    deprecation regex");
+        }
+
+        let template_string =
+            std::fs::read_to_string(path).map_err(|e| {
+                                             TemplateFileError::IOError(e, name.to_owned())
+                                         })?;
+
+        template_string.split("\n")
+            .enumerate()
+            .filter(|(_i, v)| RE.is_match(&v))
+            .for_each(|(i, v)| {
+                println!("\n\n***************************************************\n\
+                          Deprecated object access syntax in handlebars template\n\n\
+                          TEMPLATE: {}\n\
+                          LINE: {}: '{}'\n\n\
```

Comment:
I'm not showing the column because that would involve another loop, but I'm showing the complete corrected line. I think that's enough info for the user to fix the problem.

Reply:
Leaving out the column is fine. The important bit is putting the line number directly after the path, separated by a colon. For example, rustc gives warnings like:
```
warning: unused import: `validation`
  --> components/hab/src/cli.rs:23:28
```
This syntax allows editors to jump to the right line in the right file in a way that 
```
TEMPLATE: components/hab/src/cli.rs
LINE: 23
```
wouldn't. Does that make sense?

Reply:
Ah, I see. I can make that change.

Reply:
```
***************************************************
warning: Deprecated object access syntax in handlebars template
Use 'object.[index]' syntax instead of 'object[index]'
See https://github.com/habitat-sh/habitat/issues/6323 for more information

--> /hab/pkgs/core/kibana/6.1.0/20190402203539/config/kibana.yml:12

change 'elasticsearch.url: "http://{{bind.elasticsearch.members[0].sys.ip}}:{{bind.elasticsearch.members[0].cfg.http-port}}"'
to     'elasticsearch.url: "http://{{bind.elasticsearch.members.[0].sys.ip}}:{{bind.elasticsearch.members.[0].cfg.http-port}}"'

*******************************************************
```
How does this look to you?

Reply:
💯